### PR TITLE
Room Migration: Scala Compatibility

### PR DESCRIPTION
**Pull request by BradleyWire**
## What's new in this PR?

JIRA ticket: https://wearezeta.atlassian.net/browse/AN-6643

### Issues

- Room needed a migration to enable the switch from SQL to Room.
- Scala wasn't reading the database correctly

### Causes

Migrated was wrong regarding schema and names 

### Solutions

Fixed Schema and reverted names back to the original database naming

#### APK
[Download build #1208](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1208/artifact/build/artifact/wire-dev-PR2593-1208.apk)

### CODE REVIEW

File: storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabaseMigration.kt [hunk 1]
```diff
@@ -6,10 +6,10 @@ import com.waz.zclient.storage.BuildConfig
 
 private const val START_VERSION = 125
 private const val END_VERSION = 126
```
Comment (BradleyWire) [inline storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabaseMigration.kt]: We leave this at 126 for now so it doesn't interrupt the current database. But, when we go live (or when it needs testing) you'd need to bump this and the UserDatabase to 127.

**Comment by android10** [comment]
Is there a JIRA ticket you can link in the description?

**Comment by BradleyWire** [comment]
> Is there a JIRA ticket you can link in the description?

Updated the description and the JIRA ticket itself.

**Comment by BradleyWire** [comment]
build again

**Comment by BradleyWire** [comment]
build again

**Comment by BradleyWire** [comment]
build again

**Comment by BradleyWire** [comment]
build again

File: storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabaseMigration.kt [hunk 2]
```diff
@@ -6,10 +6,10 @@ import com.waz.zclient.storage.BuildConfig
...
-private const val USER_PREFERENCE_TABLE_NAME = "user_preference"
+private const val USER_PREFERENCE_TABLE_NAME = "KeyValuesTemp"
```
Comment (mejdoo) [inline storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabaseMigration.kt]: not very important but you can you unify temporary table names `KeyValuesTemp`and `UsersTemps` or `KeyValuesCopy`and `UsersCopy`.

File: storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabaseMigration.kt [hunk 3]
```diff
@@ -30,10 +29,18 @@ private const val NEW_CLIENT_LOCATION_NAME_KEY = "locationName"
...
     override fun migrate(database: SupportSQLiteDatabase) {
         if (BuildConfig.KOTLIN_SETTINGS_MIGRATION) {
-            database.execSQL("""CREATE TABLE '$NEW_CLIENT_TABLE_NAME' (
+            migrateClientTable(database)
+            migrateKeyValuesTable(database)
+            migrateUserTable(database)
+        }
+    }
+
+    //TODO still needs determining what to do with this one.
```
Comment (mejdoo) [inline storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabaseMigration.kt]: I guess we can't keep the weird structure for the table client (Json format). Think about a migration when you start implementing devices screen.

Comment (BradleyWire) [reply storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabaseMigration.kt]: 100% agree! We can improve everything at the end but in increments.

**Comment by BradleyWire** [comment]
build again

**Pull request merged**